### PR TITLE
Feature: collection section

### LINKED ISSUE #423: [Enhancement] "Collection" section
Origin: https://github.com/Soundnode/soundnode-app/issues/305

The goal is to re-create "Collection" section from SoundCloud website, which will allow to see liked artists, tracks and playlists.

To be more consistent with SoundCloud website global navigation, it will be nice to separate global menu items (like stream and collection) from user's menu items (resources created by current user - tracks, playlists, etc) in the Soundnode sidebar:

To navigate between liked entities (Artists, Tracks, Playlists) in "Collection" section one could use tabs (again like on SoundCloud website)

Question: is "Collection -> Overview" tab required? It kinda shows at max 12 items for each tab (12 artists, 12 tracks, 12 playlists) with a link to a respective tab.

### PR DESCRIPTION
**Pull request by mradionov**
Closes #423

Favorites and followings were moved to collection section (files for them too), also added liked playlists tab under collection section (uses API v2 to get liked playlists). When user clicks on a playlist, current queue gets cleared and new playlist starts playing. I was thinking about adding a separate page for a single playlist entity in another PR (I'll create an issue for that if you don't mind).
I've removed app.css.map from git index and added it to .gitignore, I guess it does not have to be under version control.
Also, I've updated angular-ui-router from 2.10 to 2.11 to make use of ui-sref advanced class activation (when route triggers).

### CODE REVIEW

File: app/public/js/collection/playlistsCtrl.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    $scope.title = 'Collection - Playlists';
+    $scope.data = '';
+    $scope.busy = false;
+
+    var clearWatch = $rootScope.$watch('userId', function (newValue, oldValue) {
+        if (newValue) {
+            clearWatch();
+            activate();
+        }
+    });
```
Comment (mradionov) [inline app/public/js/collection/playlistsCtrl.js]: Request to API requires current user id, but when controller initializes it is still undefined. Here I am waiting for a first truthy value and then activate the controller.

File: app/public/js/collection/playlistsCtrl.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    $scope.title = 'Collection - Playlists';
+    $scope.data = '';
+    $scope.busy = false;
+
+    var clearWatch = $rootScope.$watch('userId', function (newId, oldId) {
+        if (newId) {
+            clearWatch();
+            getPlaylists(newId);
+        }
+    });
```
Comment (mradionov) [inline app/public/js/collection/playlistsCtrl.js]: Request to API requires current user id, but when controller initializes it is still undefined. Waiting for it to get started

**Comment by Pitros** [comment]
For me everything seems to be working fine. What about splitting `Collections

**Comment by mradionov** [comment]
Do you mean to always show all collection sections in the sidebar menu?

**Comment by Pitros** [comment]
That's what I meant.

**Comment by weblancaster** [comment]
@mradionov this is great! 

Although I have a couple of concerns:
1. the user will need more clicks to get where they want.. in this PR the user will need to navigate to collection then whatever section they want to then play the song/track.
2. On playlists tab I can see the playlist name and how many tracks contains but what about if I just want to play a song, it's not clear to the user that they need to click the playlist then open the Queue list to choose the particular song.

I also agree with @Pitros

Let's keep discussing this to see where we can get from here.

Thanks for the hard work.

_ps: Our goal isn't per say copy Soundcloud.. but make as easier as possible for the user to use the App._

**Comment by mradionov** [comment]
Thanks for the feedback.
1. I also agree with adding sub-items to menu. I'd keep the tabs navigation though, it will give faster access to tabs. Or is it too much repetition? I think that sub-menu should be always visible, not collapsible (hides/shows on parent click). I'll experiment a bit with it to get appropriate final view.
2. It was my attempt to copy SoundCloud website. If you say that it's not the point, then at least I could re-use the same view and probably the same controller as it is on "playlists" page to be consistent. Then a separate page for a particular playlist (I was mentioning it in the PR description) is no longer a major issue. So, what do you think about having the same view and functionality for "Collection -> Playlists" and "You -> Playlists"?

**Comment by weblancaster** [comment]
@mradionov  Yeah I think it's repetitive having on the sidebar and tabs.

But I like the idea of having the playlists then a page for them containing the tracks just like any other page e.g stream, likes

**Comment by mradionov** [comment]
> But I like the idea of having the playlists then a page for them containing the tracks just like any other page e.g stream, likes

It could be a separate issue I guess - create a standalone page for a playlist. It may have a look similar to user profile page or a single track page, because playlists also have an image, description, duration and some actions, which can be applied to them.

And what do you think about my suggestion to reuse playlists view for both "Collection -> Playlists" and "You -> Playlists"?

BTW, for "Collection -> Playlists" API returns both liked and user-owned playlists, now I am showing everything. It is possible to filter them and show only liked or only user-owned. This filter can be presented in user interface too, so user could switch between his own playlists and liked playlists. If we do so, then we won't need two playlist pages. Even more, I guess we can revert collection section, because the point of it for me was to separate to different playlist pages.

**Comment by mradionov** [comment]
My update on this: I've decided to use a view similar to existing playlists and reorganized sidebar menu a bit. Because logic and available functions are a bit different I am using different controller and view for collection playlists. Is it a way to go?

image

**Comment by mradionov** [comment]
Just a friendly reminder, that this PR is available for review again after my reworks, I forgot to leave a note about it when I pushed an update. Thanks!

**Comment by weblancaster** [comment]
I will do thanks @mradionov

**Comment by weblancaster** [comment]
Hey @mradionov the PR looks good but I can't play the tracks under `collection > playlists`

**Comment by mradionov** [comment]
@weblancaster Sorry, I do not quite get your idea about embedding playlists on a stream view, could you please explain what do you mean by it. Thanks

**Comment by weblancaster** [comment]
Have them in the same view `stream` in this case.

**Comment by mradionov** [comment]
Like move a menu item? 
image

I kinda agree that it is confusing having to "Playlists". Maybe just rename them?

I was trying to explain one more idea in my comment above - to have one "Playlists" page as it was before. Menu will remain untouched (no splitting to sections "main", "collection", "me"). The only difference will be on playlists page, where user will be able to switch between two lists of "liked" and "owned" playlists. Is it ok or is it an extra steps (extra click) like it was with tabs? 

image

Or maybe display "liked" and "owned" in one list, but mark respective playlist with an icon to show if it's "liked" or "owned".

image

What do you think about it in comparison with my previous versions?

**Comment by weblancaster** [comment]
@mradionov I will close this issue since we are moving on. Thanks for the work though but I think we will keep as it is.

**Pull request closed without merging**